#164829201 send user verification link via email

Sm0k3s opened the pull request:
#### Description
When a user signs up an email is instantly sent to the email they provided. It contains the verification link that is unique to them and can only be used once. 

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [x] End to end
- [x] Integration

#### Checklist:
- [ ] My code follows the style guide for this project
- [ ] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[#164829201](https://www.pivotaltracker.com/n/projects/2321088/stories/164829201)

ValerieRono committed:
feature(email verification)-send user verification link via email

- add functionality to send email upon user sign
- add tests for the new feature

[Delivers #164829201]

ValerieRono committed:
feature(email verification)-send user verification link via email

- add functionality to send email upon user sign
- add tests for the new feature

[Delivers #164829201]

ValerieRono reviewed (approved):
looks good to me Steve

ogiste commented on authors/apps/authentication/views.py:
```diff
@@ -24,9 +32,49 @@ def post(self, request):
+        if user is not None and email_activation_token.check_token(user, token):
+            user.email_confirmed = True
+            user.save()
+            return Response("Email successfully verified")
```
I would _suggest_ adding key for the message - "Email successfully verified" and any other response messages you send.

Sm0k3s replied:
Let me work on it.

Sm0k3s committed:
feature(email verification)-send user verification link via email

- add functionality to send email upon user sign-up
- add tests for email verification
- add template for email

[Delivers #164829201]

koitoror reviewed (changes requested):
squash commits to 1

Sm0k3s committed:
feature(email verification)-send user verification link via email

- add tests for email verification
- fixes failing tests
- add template for email

[Delivers #164829201]

ogiste reviewed (approved):
Tests are passing and request body format has also been updated. Looks good.
